pkg/dazzle/combiner.go

```diff
@@ -311,14 +311,21 @@ func mergeEnv(base *ociv1.Image, others []*ociv1.Image, vars []EnvVarCombination
 				case EnvVarCombineMerge:
 					envs[k] += ":" + v
 				case EnvVarCombineMergeUnique:
-					vs := make(map[string]struct{})
-					for _, s := range strings.Split(envValue, ":") {
-						vs[s] = struct{}{}
+					var vs []string
+					vs = append(vs, strings.Split(envValue, ":")...)
+					vs = append(vs, strings.Split(v, ":")...)
+
+					vss := []string{}
+					flags := make(map[string]bool)
+					for i := len(vs) - 1; i >= 0; i-- {
+						entry := vs[i]
+						if _, value := flags[entry]; !value {
+							flags[entry] = true
+							vss = append(vss, entry)
+						}
 					}
-					vs[v] = struct{}{}
-					vss := make([]string, 0, len(vs))
-					for s := range vs {
-						vss = append(vss, s)
+					for i, j := 0, len(vss)-1; i < j; i, j = i+1, j-1 {
+						vss[i], vss[j] = vss[j], vss[i]
```

Comment:
The original code is simple to modify for sort order. All it takes is a `sort.Strings` before joining the values
```Go
sort.Strings(vss)
envs[k] = strings.Join(vss, ":")
```